### PR TITLE
GDPR Privacy Information links (batch 1)

### PR DESCRIPTION
**Pull request by jaswrks**
~~This PR depends on https://github.com/Automattic/wp-calypso/pull/24985.~~ (merged).

The original full PR was broken apart by request: https://github.com/Automattic/wp-calypso/pull/23949#issuecomment-382109504.

---

This is batch one of the new Privacy Information links, implemented using the new `SupportInfo` component in https://github.com/Automattic/wp-calypso/pull/24985

## Overview of Changes

- Updates CSS, changing `.site-settings__info-link-container` to `.support-info`.

- Removes `.site-settings__info-link-container` in favor of SupportInfo.

- Adds support info icons to `JetpackModuleToggle` instances, covering every Jetpack module/feature. In batch one I covered the following Jetpack features:

  - Comment Display Settings
  - After the Deadline
  - Custom Content Types
  - Google Analytics
  - Comment Links + Hovercards
  - Jetpack Monitoring

### CODE REVIEW

**Comment by alisterscott** [comment]
This also isn't building in calypso.live

**Comment by jaswrks** [comment]
This PR was broken apart by request (https://github.com/Automattic/wp-calypso/pull/23949#issuecomment-382109504).

To test using calypso.live, please see: https://github.com/Automattic/wp-calypso/pull/23949
https://calypso.live/?branch=add/gdpr-feature-privacy-links

**Comment by Copons** [comment]
This looks tentatively good to me, but I'd rather test it after it's rebased on #24985.

One thing that looks a bit off is how the strings are split. See for example:
https://github.com/Automattic/wp-calypso/pull/24987/files#diff-3ae2d6fca2ce5da114eb4de111ed2287R220

@jaswrks could you make sure they _need_ to be split, and if they don't (like in the case above) remove the split?

The idea is to have a max line length of `100`, but we recently removed all ESLint formatting rules, only relying on Prettier, which does a lot of good things but doesn't break strings longer than the max-len limit.
In other words, right now we both don't have ESLint telling us there's something wrong, and Prettier fixing it, so we need to fix these cases manually.

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 1]
```diff
@@ -216,22 +215,12 @@ class AfterTheDeadline extends Component {
...
-								target="_blank"
-							>
-								{ translate( 'Learn more' ) }
-							</ExternalLink>
-						</InfoPopover>
-					</div>
-
+					<SupportInfo
+						text={ translate(
+							'Checks your content for correct grammar and spelling, misused words, and style.'
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Checks your content for correct grammar and spelling, misused words, and style while you write.' )` *ES Score: 16*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (Copons) [reply client/my-sites/site-settings/composing/after-the-deadline.jsx]: Yup, I'm with the bot on this one 😄

Comment (jaswrks) [reply client/my-sites/site-settings/composing/after-the-deadline.jsx]: Done, ty :-)

File: client/my-sites/site-settings/style.scss [hunk 2]
```diff
@@ -436,11 +436,7 @@
 	margin: 16px 36px 0;
 }
 
-.site-settings__info-link-container {
-	float: right;
```
Comment (Copons) [inline client/my-sites/site-settings/style.scss]: This creates a bunch of oddities all over Settings.

| Before | After |
| --- | --- |
| screen shot 2018-05-24 at 17 32 00 | screen shot 2018-05-24 at 17 32 08 |

**Comment by Copons** [comment]
One thing I've noticed now that this rebased, is that a [style rule](https://github.com/Automattic/wp-calypso/blob/08a96b508d34c07c0903f1bdd656bf12b0bbb9af/client/components/support-info/style.scss#L3) in #24985 causes a `SupportInfo` to push other `SupportInfo` to the left if they're close enough (if their lines are not tall enough):

screen shot 2018-05-24 at 17 25 24

**Comment by Copons** [comment]
@jaswrks This looks in a really good place already!
I've pointed out a couple of visual glitches that I think have fairly quick fixes (unless these fixes cause all kinds of regressions, of course 😄).

File: client/my-sites/site-settings/style.scss (hunk 2 above)
Comment (jaswrks) [reply client/my-sites/site-settings/style.scss]: Thank you. I believe the reason is because this PR is just batch one. Batch 2 and 3 will change all of those other icons, which effectively eliminates all use of `.site-settings__info-link-container` once they're all in.

Therefore, we'll need to review and merge all three together and deploy at the same time. Other ideas?

**Comment by jaswrks** [comment]
@Copons writes...

> One thing I've noticed now that this rebased, is that a style rule in #24985 causes a SupportInfo to push other SupportInfo to the left if they're close enough (if their lines are not tall enough):

Great catch! Thank you.

I fixed this by dropping the bottom margin on the `SupportInfo` icon. It wasn't absolutely necessary, and by not using it we avoid this problem.

**Comment by Copons** [comment]
Actually!
When you split the original PR, the line that prevented tests to fail here ended up in batch 3: https://github.com/Automattic/wp-calypso/pull/24989/files#r190851385

Mind moving it here to avoid breaking tests during deploy? 🙇

**Comment by jaswrks** [comment]
Awesome, thanks for all your help on this!

> Mind moving it here to avoid breaking tests during deploy?

Done. I'm awaiting passing tests and then I'll merge/deploy.
